documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,303 @@
+= Quarkus - Using the Cassandra Client
+
+Cassandra is a well known NoSQL Database that is widely used.
+
+In this guide, we see how you can get your REST services to use the Cassandra database.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* GraalVM installed with `GRAALVM_HOME` configured appropriately if you want to use the native mode.
+* Apache Maven 3.5.3+
+* Cassandra installed or Docker installed
+
+== Architecture
+
+The application built in this guide is quite simple: the user can add elements in a list using a form and the list is updated.
+
+All the information between the browser and the server is formatted as JSON.
+
+The elements are stored in the Cassandra Database.
+
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+The solution is located in the `quickstart` {quickstart-clone-url}[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new maven project and copy the `pom.xml` file that is present in the `quickstart`.
+
+The `pom.xml` is importing the RESTEasy/JAX-RS, JSON-B, Context Propagation and Cassandra Client extensions.
+We will be building a REST application that is using DataStax Object Mapper. It will simplify the Data Access Layer code.
+
+The most important part of the `pom.xml` is adding the `cassandra-quarkus` extension:
+[source,xml]
+----
+<dependency>
+  <groupId>com.datastax.oss.quarkus</groupId>
+  <artifactId>cassandra-quarkus-client</artifactId>
+  <version>${quarkus.version}</version>
+</dependency>
+----
+
+== Creating JSON REST service
+In this example, we will create an application to manage a list of fruits.
+
+First, let's create the `Fruit` bean as follows:
+[source,java]
+----
+@Entity
+public class Fruit {
+
+  @ClusteringColumn private String name;
```

Comment:
My idea was that `id` is a `store_name`:
`@PartitionKey private String id;`
and the REST API is inserting and retrieving data only for one particular store name. It is solving problem of fetch all. I can change the id column to `store_name`, wdyt?

Reply:
But you need to change the quickstart project too, right? I agree that `store_name` is better, but since it's not related to this PR, how about doing it in a separate PR?

Reply:
Ok, I can do followup PR with that change.
I've created a JIRA task to not forget about it:
https://datastax-oss.atlassian.net/browse/JAVA-2767

Reply:
Thank you!!